pkg/http/handlers/base.go

```diff
@@ -0,0 +1,115 @@
+package handlers
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/contiamo/go-base/pkg/errors"
+	"github.com/contiamo/go-base/pkg/models"
+	"github.com/contiamo/go-base/pkg/tracing"
+)
+
+const (
+	maxRequestBodySize = 1024 * 1024 // 1MB
+)
+
+// BaseHandler contains all the base functions every handler should have
+type BaseHandler interface {
+	tracing.Tracer
+	// Read reads the body and tries to decode JSON from it to the given output type
```

Comment:
The 1MB limit should probably be documented here

Reply:
I'll add the limit as a parameter.